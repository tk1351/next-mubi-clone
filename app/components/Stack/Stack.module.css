.stack {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.stack > * {
    margin-top: 0;
    margin-bottom: 0;
}

.stack-non-recursive > * + * {
    margin-top: 1.5rem;
}

.stack-recursive * + * {
    margin-top: 1.5rem;
}

