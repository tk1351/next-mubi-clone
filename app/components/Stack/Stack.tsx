import { type FC, type PropsWithChildren } from "react";
import clsx from "clsx";
import styles from "./Stack.module.css";

export type Props = PropsWithChildren<{
  isRecursive?: boolean;
}>;

export const Stack: FC<Props> = ({ children, isRecursive = false }) => {
  const stackClassName = clsx(styles.stack, {
    [styles["stack-recursive"]]: isRecursive,
    [styles["stack-non-recursive"]]: !isRecursive,
  });
  return <div className={stackClassName}>{children}</div>;
};
