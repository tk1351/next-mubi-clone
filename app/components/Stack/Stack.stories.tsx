import type { Meta, StoryObj } from "@storybook/react";
import { Stack } from "./Stack";

const meta = {
  title: "Components/Stack",
  component: Stack,
} satisfies Meta<typeof Stack>;

export default meta;
type Story = StoryObj<typeof meta>;
export const Primary: Story = {
  args: {
    children: (
      <>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
      </>
    ),
    isRecursive: false,
  },
};

export const Recursive: Story = {
  args: {
    children: (
      <>
        <div style={{ width: "100%", border: "2px solid purple" }}>
          <div
            style={{
              width: "100%",
              height: "30px",
              border: "1px solid orange",
            }}
          ></div>
          <div
            style={{
              width: "100%",
              height: "30px",
              border: "1px solid orange",
            }}
          ></div>
        </div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
        <div
          style={{ width: "100%", height: "30px", border: "2px solid purple" }}
        ></div>
      </>
    ),
    isRecursive: true,
  },
};
