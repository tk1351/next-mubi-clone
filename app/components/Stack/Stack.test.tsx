import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Stack } from "./Stack";

describe(`Testing Stack Component`, () => {
  describe("Snapshot", () => {
    it("isRecursive が true の場合", () => {
      const { container } = render(<Stack isRecursive={true}>Stack</Stack>);
      expect(container).toMatchSnapshot();
    });
    it("isRecursive が false の場合", () => {
      const { container } = render(<Stack isRecursive={false}>Stack</Stack>);
      expect(container).toMatchSnapshot();
    });
  });
});
