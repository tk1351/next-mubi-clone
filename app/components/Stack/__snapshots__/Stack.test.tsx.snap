// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Stack Component > Snapshot > isRecursive が false の場合 1`] = `
<div>
  <div
    class="_stack_7c767f _stack-non-recursive_7c767f"
  >
    Stack
  </div>
</div>
`;

exports[`Testing Stack Component > Snapshot > isRecursive が true の場合 1`] = `
<div>
  <div
    class="_stack_7c767f _stack-recursive_7c767f"
  >
    Stack
  </div>
</div>
`;
